From: Michael Olbrich <m.olbrich@pengutronix.de>
Date: Sun, 4 Dec 2022 13:18:18 +0100
Subject: [PATCH] qtbase: fix building with mesa 22.3.0

X11 support is now disabled by default in the EGL headers.
So explicitly enable it when QT_EGL_NO_X11 is not defined.

Signed-off-by: Michael Olbrich <m.olbrich@pengutronix.de>
---
 qtbase/src/gui/configure.json                        | 3 ++-
 qtbase/src/platformsupport/eglconvenience/qt_egl_p.h | 2 ++
 2 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/qtbase/src/gui/configure.json b/qtbase/src/gui/configure.json
index 1f08795c57b4..12c95742d2da 100644
--- a/qtbase/src/gui/configure.json
+++ b/qtbase/src/gui/configure.json
@@ -834,7 +834,8 @@
                     "// embedded devices, are not intended to be used together with X. EGL support",
                     "// has to be disabled in plugins like xcb in this case since the native display,",
                     "// window and pixmap types will be different than what an X-based platform",
-                    "// plugin would expect."
+                    "// plugin would expect.",
+                    "#define USE_X11"
                 ],
                 "include": [ "EGL/egl.h", "X11/Xlib.h" ],
                 "main": [
diff --git a/qtbase/src/platformsupport/eglconvenience/qt_egl_p.h b/qtbase/src/platformsupport/eglconvenience/qt_egl_p.h
index bf37d07fd805..1dff731e7f78 100644
--- a/qtbase/src/platformsupport/eglconvenience/qt_egl_p.h
+++ b/qtbase/src/platformsupport/eglconvenience/qt_egl_p.h
@@ -61,6 +61,8 @@
 # if !defined(Q_OS_INTEGRITY)
 #  define WIN_INTERFACE_CUSTOM   // NV
 # endif // Q_OS_INTEGRITY
+#else  // QT_EGL_NO_X11
+# define USE_X11
 #endif  // QT_EGL_NO_X11
 
 #ifdef QT_EGL_WAYLAND
